flink/v1.18/flink/src/main/java/org/apache/iceberg/flink/FlinkSchemaUtil.java

```diff
@@ -64,26 +68,75 @@ public static Schema convert(TableSchema schema) {
     RowType root = (RowType) schemaType;
     Type converted = root.accept(new FlinkTypeToType(root));
 
-    Schema iSchema = new Schema(converted.asStructType().fields());
-    return freshIdentifierFieldIds(iSchema, schema);
+    Schema icebergSchema = new Schema(converted.asStructType().fields());
+    return freshIdentifierFieldIds(icebergSchema, schema);
   }
 
-  private static Schema freshIdentifierFieldIds(Schema iSchema, TableSchema schema) {
+  /** Convert the flink table schema to apache iceberg schema with column comment. */
+  public static Schema convert(ResolvedSchema flinkSchema) {
+    List<Column> tableColumns = flinkSchema.getColumns();
+    // copy from org.apache.flink.table.api.Schema#toRowDataType
+    DataTypes.Field[] fields =
+        tableColumns.stream()
+            .map(
+                column -> {
+                  if (column.getComment().isPresent()) {
+                    return DataTypes.FIELD(
+                        column.getName(), column.getDataType(), column.getComment().get());
+                  } else {
+                    return DataTypes.FIELD(column.getName(), column.getDataType());
+                  }
+                })
+            .toArray(DataTypes.Field[]::new);
+
+    LogicalType schemaType = DataTypes.ROW(fields).notNull().getLogicalType();
+    Preconditions.checkArgument(
+        schemaType instanceof RowType, "Schema logical type should be RowType.");
+
+    RowType root = (RowType) schemaType;
+    Type converted = root.accept(new FlinkTypeToType(root));
+    Schema icebergSchema = new Schema(converted.asStructType().fields());
+    return freshIdentifierFieldIds(icebergSchema, flinkSchema);
+  }
+
+  /** @deprecated Use {@link #freshIdentifierFieldIds(Schema, ResolvedSchema)} instead. */
+  @Deprecated
+  private static Schema freshIdentifierFieldIds(Schema icebergSchema, TableSchema flinkSchema) {
```

Comment:
the diff looks a little odd to me :) the old function looks as new code, while the function appears to be new code. just cosmetic thing though

Reply:
It does look a bit strange. This method just takes out the primary key of flink and sets the non-empty column of iceberg, so I changed the input parameters. Can you see if this is reasonable?